src/lib.rs

```diff
@@ -9,8 +10,14 @@ pub struct Mod {
 
 impl Mod {
     pub fn new(value: i32, modulus: i32) -> Mod {
+        let remainder = value % modulus;
+        let value = match remainder > 0 {
```

Comment:
Shouldn't it be `>= 0` ? If `remainder == 0`, the `value` will equal `modulus`. Is it wanted ?